run-checks

```diff
@@ -111,41 +117,32 @@ if [ ! -z "$UNIT" ]; then
     mkdir .coverage
     echo "mode: set" > .coverage/coverage.out
 
-    echo Installing godeps
-    go get launchpad.net/godeps
-    export PATH=$PATH:$GOPATH/bin
-
-    echo Obtaining dependencies
-    godeps -u dependencies.tsv
-
     echo Building
-    go build -v github.com/ubuntu-core/snappy/...
+    go build -tags=excludeintegration -v github.com/ubuntu-core/snappy/...
 
     # tests
     echo Running tests from $(pwd)
-    for pkg in $(go list ./...); do
-        $goctest -v -coverprofile=.coverage/profile.out $pkg
+    for pkg in $(go list ./... | grep -v integration-tests); do
+        $goctest -tags=excludeintegration -v -coverprofile=.coverage/profile.out $pkg
         append_coverage .coverage/profile.out
     done
 
     # the rabbit hole
+    echo Building the integration tests
+    go build integration-tests/main.go
+
     echo Running the tests for the integration testutils
-    # go list doesn't take into account build tags
-    for pkg in $(ls ./integration-tests/testutils/); do
-        $goctest -tags=integration -v -coverprofile=.coverage/profile.out "github.com/ubuntu-core/snappy/integration-tests/testutils/$pkg"
+    for pkg in $(go list ./integration-tests/testutils/...); do
+        $goctest -v -coverprofile=.coverage/profile.out $pkg
         append_coverage .coverage/profile.out
     done
 fi
 
 if [ ! -z "$INTEGRATION" ]; then
-    # integration tests
-    echo Building the integration tests
-    go build -tags=integration integration-tests/main.go
-
     # integration suite in kvm
     if which adt-run >/dev/null 2>&1; then
         echo "Running integration tests on rolling edge"
-        go run -tags=integration integration-tests/main.go --snappy-from-branch
+        go run integration-tests/main.go --snappy-from-branch
         # print the results.
         if which subunit2pyunit >/dev/null 2>&1; then
             subunit-1to2 /tmp/snappy-test/output/artifacts/results.subunit | subunit2pyunit
```

Comment:
Another nit, don't know from where exactly but after executing the checks I'm getting a leftover "main" executable in the root of the project, we could get rid of it